client/src/components/EditBox.js

```diff
@@ -29,30 +22,76 @@ export default function EditBox() {
         console.log(data);
         setLearningObj(data);
       });
+  };
+  useEffect(() => {
+    getLearningObj();
   }, [id]);
 
-  const addLearningObjective = (description) => {
-    setLearningObjective(skills.push(description));
-  };
+  // const deleteLearningOb = (LearningID)=> {
+  //   console.log(LearningID)
+  //       fetch(`/api/learningobjectives/${LearningID}`, {
+  //           method: "DELETE",
+  //           headers: {
+  //             'Content-Type': 'application/json',
+  //           },
+  //         })
+  //           .then(res => res.json())
+  //           .then((data) =>data);
 
-  const deleteLearningObjective = (objective) => {
-    let index = skills.indexOf(objective);
-    if (index > -1) {
-      setDeleted(skills.splice(index, 1));
+  //       }
+
+  const updateLearningObj = (description, LearningID) => {
+    console.log(description);
+    if (description) {
+      fetch(`/api/learningobjectives/${LearningID}`, {
+        method: "PUT",
+        body: JSON.stringify({
+          description,
+        }),
+        headers: {
+          "Content-Type": "application/json",
+        },
+      })
+        .then((res) => res.json())
+        .then((data) => {
+          setUpdateLO("");
+          getLearningObj();
+        });
     }
+    return;
   };
 
+  const cancelUpdate = (id) =>
+    new Promise((resolve) => {
+      resolve(id);
+    }).then(() => {
+      setUpdateLO("");
+      getLearningObj();
+    });
+
+  const addLearningObjective = (description) => {
+    setLearningObjective(skills.push(description));
+  };
   return (
     <div className="learning-objective-container">
       <h2 className="skill-name">{id}</h2>
       <div>
         <ul>
-          {learningObj.map(({ description }, index) => {
+          {learningObj.map(({ description, id }, index) => {
             return (
               <li key={index}>
-                <div>{description}</div>
                 <div className="edit-delete-buttons">
-                  <button
+                  
+                 
+                  {updateLO === index ? (
+              <input
+                className="app-message__input"
+                onChange={e => setUpdateLO(e.target.value)}
+                value={learningObj.description}
+              ></input>
+            ) : (
```

Comment:
Adding this `updateLO === index` seems to make the component a bit confusing here, especially as you're already using another ternary on line 106 updateLO === id. But it's good that this works, it can always be refactored to improve readability later on 👍